react/features/base/dialog/components/Dialog.web.js

```diff
@@ -0,0 +1,161 @@
+import React from 'react';
+import { connect } from 'react-redux';
+import AKButton from '@atlaskit/button';
+import AKButtonGroup from '@atlaskit/button-group';
+import ModalDialog from '@atlaskit/modal-dialog';
+
+import AbstractDialog, { _mapStateToProps } from './AbstractDialog';
+import { translate } from '../../i18n';
+
+/**
+ * Web dialog that uses atlaskit modal-dialog to display dialogs.
+ */
+class Dialog extends AbstractDialog {
+
+    /**
+     * Web dialog component's property types.
+     *
+     * @static
+     */
+    static propTypes = {
+        /**
+         * This is the body of the dialog, the component children.
+         */
+        children: React.PropTypes.node,
+
+        /**
+         * Whether the dialog is modal. This means clicking on the blanket will
+         * leave the dialog open. No cancel button.
+         */
+        isModal: React.PropTypes.bool,
+
+        /**
+         * Width of the dialog, can be:
+         * - 'small' (400px), 'medium' (600px), 'large' (800px),
+         * 'x-large' (968px)
+         * - integer value for pixel width
+         * - string value for percentage
+         */
+        width: React.PropTypes.string
+    }
+
+    /**
+     * Implements React's {@link Component#render()}.
+     *
+     * @inheritdoc
+     * @returns {ReactElement}
+     */
+    render() {
+        return (
+            <ModalDialog
+                footer = { this._renderFooter() }
+                header = { this._renderHeader() }
+                isOpen = { this.props.show }
+                onDialogDismissed = { this._onCancel }
+                width = { this.props.width
+                    ? this.props.width : 'medium' }>
+                <div>
+                    <form
+                        id = 'modal-dialog-form'
+                        onSubmit = { this._onSubmit }>
+                        { this.props.children }
+                    </form>
+                </div>
+            </ModalDialog>);
+    }
+
+    /**
+     * Render cancel button.
+     *
+     * @returns {*} The cancel button if enabled and dialog is not modal.
+     * @private
+     */
+    _renderCancelButton() {
+        if (this.props.cancelDisabled || this.props.isModal) {
```

Comment:
Yes. It doesn't work if we have. The thing is that clicking outside dialog is the same as clicking cancel, so modality will not work if having a cancel. This is limitation of atlaskit modal dialog. I discussed this with @yanas and she agreed with this approach.